src/Angor/Avalonia/Angor.Contexts.Funding/Investor/Domain/PortfolioService.cs

```diff
@@ -83,6 +83,38 @@ public async Task<Result> Add(Guid walletId, InvestmentRecord newInvestment)
             : Result.Failure("Failed to save investment record");
     }
 
+    public async Task<Result> Update(Guid walletId, InvestmentRecord updatedInvestment)
```

Comment:
This code does exactly the same as the code of the add method, better to just call the Add Upsert and remove this code